src/upload.cpp

```diff
@@ -404,6 +588,43 @@ bool Upload::validateTeleTANCharacters(QString const &teleTAN) const
     return result;
 }
 
+bool Upload::validateGUID(QString const &guid) const
+{
+    bool result;
+
+    qDebug() << "Validating GUID: " << guid;
+    result = guid.length() == GUID_LENGTH;
+
+    qDebug() << "Valid Length: " << result;
+
+    for (uint16_t pos = 0; pos < GUID_LENGTH && result; ++pos) {
+        result &= validateGUIDCharacter(guid.at(pos));
+        qDebug() << "Validating GUID Character: " << guid.at(pos) << " valid: " << result;
+    }
+    return result;
+}
+
+bool Upload::validateGUIDCharacters(QString const &guid) const
+{
+    int pos;
+    bool result;
+
+    result = true;
+    pos = 0;
+    while (pos < 10 && pos < guid.length() && result) {
```

Comment:
```suggestion
    while (pos < GUID_LENGTH && pos < guid.length() && result) {
```

Reply:
:( Embarassing mistake after I finally realised why you chose a while loop and added it back again, Thx!